Source/Particle/NyxParticleContainer.H

```diff
@@ -66,9 +66,16 @@ public:
     
     void GetParticleVelocities (amrex::Vector<amrex::Real>& part_vels);
     void SetParticleVelocities (amrex::Vector<amrex::Real>& part_data);
-    
+
     virtual amrex::Real sumParticleMass (int level) const override {
-        return amrex::NeighborParticleContainer<NSR,NSI>::sumParticleMass(0,level);
+        using PType = typename amrex::ParticleContainer<NSR, NSI>::SuperParticleType;
+        auto msum = amrex::ReduceSum(*this, level,
+                                     [=] AMREX_GPU_HOST_DEVICE (const PType& p) -> amrex::Real
+                                     {
+                                         return p.rdata(0);
```

Comment:
For consistency with previous amrex implementation, we should check validity of the particle
https://github.com/AMReX-Codes/amrex/blob/fd4e0d1078a489fe9f07f0d4aa9de77f1680ed2c/Src/Particle/AMReX_ParticleContainerI.H#L698
`if(p.id()>0)
return p.rdata(0);
else
return 0;`